File: app/src/main/java/com/moroom/android/data/model/Review.java
package com.moroom.android.data.model;

import java.util.HashMap;

public class Review {
    private String title = "";
    private String pros = "";
    private String cons = "";
    private String checkedItems = "";

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getPros() {
        return pros;
    }

    public void setPros(String pros) {
        this.pros = pros;
    }

    public String getCons() {
        return cons;
    }

    public void setCons(String cons) {
        this.cons = cons;
    }

    public String getCheckedItems() { return checkedItems; }

    public void setCheckedItems(String checkedItems) { this.checkedItems = checkedItems; }
}

File: app/src/main/java/com/moroom/android/data/source/remote/model/Review.kt
package com.moroom.android.data.source.remote.model

import com.google.firebase.auth.FirebaseAuth
import com.moroom.android.domain.model.WrittenReview
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

data class Review(
    val address: String = "",
    val title: String = "",
    val pros: String = "",
    val cons: String = "",
    val checkedItems: String = "",
    val idToken: String = "",
    val createdAt: String = SimpleDateFormat("yyyy-MM-dd HH:mm", Locale.getDefault()).format(Date())
)

fun WrittenReview.toDataModel(auth: FirebaseAuth): Review = Review(
    address = this.address,
    title = this.floor + " " + this.year + "년도까지 거주 " + this.rentType,
    pros = this.pros,
    cons = this.cons,
    checkedItems = this.checkItems.filter { it.isChecked }.joinToString(separator = "\n") { it.listText },
    idToken = auth.uid!!
)